model/common/src/icon4py/model/common/orchestration/halo_exchange.py

```diff
@@ -0,0 +1,325 @@
+
+            {fields_desc}
+
+            h_{unique_id} = communication_object->exchange({", ".join([f'(*pattern)({"" if wait else "*"}{descr_unique_names[i]})' for i in range(len(global_buffers))])});
+            { 'h_'+str(unique_id)+'.wait();' if wait else ''}
+
+            __out = 1234; // Dummy return;
+            """
+
+    tasklet.code = CodeBlock(code=code, language=dace.dtypes.Language.CPP)
+
+    if counter == 0:
+        # Set global variables
+        __pattern = ""
+        __domain_descriptor = ""
+        for dim_ in dims.global_dimensions.values():
+            __pattern += f"{dace.uintp.dtype} __pattern_{dim_.value}Dim_ptr_{unique_id};\n"
+            __domain_descriptor += (
+                f"{dace.uintp.dtype} __domain_descriptor_{dim_.value}Dim_ptr_{unique_id};\n"
+            )
+
+        code = f"""
+                {dace.uintp.dtype} __context_ptr_{unique_id};
+                {dace.uintp.dtype} __comm_ptr_{unique_id};
+                {__pattern}
+                {__domain_descriptor}
+                {fields_desc_glob_vars}
+                ghex::communication_handle<{communication_handle_type}> h_{unique_id};
+                """
+    else:
+        code = fields_desc_glob_vars
+
+    tasklet.code_global = CodeBlock(code=code, language=dace.dtypes.Language.CPP)
+
+
+@dace.library.environment
+class DaceGHEX:
+    """Set GHEX environment for compilation in DaCe"""
+
+    python_site_packages: ClassVar[str] = site.getsitepackages()[0]
+    ghex_path: ClassVar[str] = (
+        python_site_packages + "/ghex"
+    )  # 'absolute_path_to/GHEX/build' [case of manual compilation]
+
+    cmake_minimum_version: ClassVar = None
+    cmake_packages: ClassVar = []
+    cmake_variables: ClassVar = {}
+    cmake_compile_flags: ClassVar = []
+    cmake_link_flags: ClassVar[list[str]] = [
+        f"-L{ghex_path}/lib -lghex -lhwmalloc -loomph_common -loomph_mpi"
+    ]
+    cmake_includes: ClassVar[list[str]] = [
+        f"{sys.prefix}/include",
+        f"{ghex_path}/include",
+        f"{python_site_packages}/gridtools_cpp/data/include",
+    ]
+    cmake_files: ClassVar = []
+    cmake_libraries: ClassVar = []
+
+    headers: ClassVar[list[str]] = [
+        "ghex/context.hpp",
+        "ghex/unstructured/pattern.hpp",
+        "ghex/unstructured/user_concepts.hpp",
+        "ghex/communication_object.hpp",
+        "stdexcept",
+    ]
+    state_fields: ClassVar = []
+    init_code: ClassVar = ""
+    finalize_code: ClassVar = ""
+    dependencies: ClassVar = []
+
+
+class DummyNestedSDFG:
+    def __sdfg__(self, *args, **kwargs) -> dace.SDFG:
+        sdfg = dace.SDFG("DummyNestedSDFG")
+        state = sdfg.add_state()
+
+        sdfg.add_scalar(name="__return", dtype=dace.int32)
+
+        tasklet = dace.sdfg.nodes.Tasklet(
+            "DummyNestedSDFG",
+            inputs=None,
+            outputs=None,
```

Comment:
My experience tells that `outputs` should be the set of output connectors: `{"__out"}`

Reply:
you are right. This is just an initialisation of the tasklet. If you check a bit below, I set it properly.
```
out_connectors["__out"] = ...
tasklet.out_connectors = out_connectors
```